dist/d3-flamegraph.js

```diff
@@ -5211,17 +5212,100 @@ var flamegraph = function () {
     });
   }
 
-  function s4 () {
-    return Math.floor((1 + Math.random()) * 0x10000)
-      .toString(16)
-      .substring(1)
-  }
+  function injectIds (node, parentId, rank, nSibs, depth) {
```

Comment:
The id generating works, creating unique ids, but it's extremely heavy. With a _not that large_ sample file, this solution is ~10x slower than the random id solution.